lib/portlayer/storage/vsphere/datastore.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vsphere
+
+import (
+	"io"
+	"path"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	"golang.org/x/net/context"
+)
+
+// Datastore gives access to the datastore regardless of type (esx, esx + vc,
+// or esx + vc + vsan).  Also wraps paths to a given root directory
+type datastore struct {
```

Comment:
Yeah, i struggled with this a bit.  I didn't want this to be confused with the govmomi `object.Datastore` and I didn't want to export it to be used outside this package.  So, I made the struct private to the package, made the `new` method private, and only use the exported functions outside this file (except for the 1 place which I commented).  

Yeah TL;DR-  i just didnt want to export this since it doesn't do much and was painful to get right.